internal/utils/flags/flags.go

```diff
@@ -0,0 +1,122 @@
+package flags
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/spf13/pflag"
+)
+
+const (
+	//TypeString is the type of strings
+	TypeString = "string"
+)
+
+// enumSetValue is modified copy of stringSliceValue from
+// the cobra pflags package
+type enumSetValue struct {
+	value          *[]string
+	changed        bool
+	validValuesSet map[string]bool
+}
+
+// NewEnumSetValue creates a stringSliceValue that validates inputs and prevents duplicates
+func NewEnumSetValue(p *[]string, val []string, validValues []string) pflag.Value {
```

Comment:
can we just return an exported type of `*EnumSetValue` here?  I like the idea here, but with the fields already un-exported we aren't gaining much by returning the interface here (and only half of the implemented interfaces since we've omitted `SliceValue` here)

also, can we just call this `NewEnumSet`?  I think the `flags` package prefix gives us enough context into what we're creating here is actually a flag value